### PR TITLE
[TD-11626]<feature>taosdemo support pressure test compare with perfMonitor

### PR DESCRIPTION
**Pull request by zhaoyanggh**
only in taosc progressive mode

### CODE REVIEW

**Review by afwerar — APPROVED** [review]
LGTM

File: src/demoInsert.c [hunk 1]
```diff
@@ -2845,6 +2847,15 @@ void *syncWrite(void *sarg) {
 
 int startMultiThreadInsertData(int threads, char *db_name, char *precision,
                                SSuperTable *stbInfo) {
+    if (stbInfo && g_args.pressure_mode) {
+      stbInfo->buffer = calloc(1, stbInfo->columnCount*2 + 2 );
```
Comment (sangshuduo) [inline src/demoInsert.c:2851]: indent

File: src/demoInsert.c [hunk 2]
```diff
@@ -2582,28 +2583,29 @@ void *syncWriteProgressive(threadInfo *pThreadInfo) {
...
+              percentComplete = currentPercent;
             }
             int64_t currentPrintTime = taosGetTimestampMs();
             if (currentPrintTime - lastPrintTime > 30 * 1000) {
-                printf("thread[%d] has currently inserted rows: %" PRId64
-                       ", affected rows: %" PRId64 "\n",
-                       pThreadInfo->threadID, pThreadInfo->totalInsertRows,
-                       pThreadInfo->totalAffectedRows);
-                lastPrintTime = currentPrintTime;
+              printf("thread[%d] has currently inserted rows: %" PRId64
```
Comment (sangshuduo) [inline src/demoInsert.c:2600]: indent

File: src/demoInsert.c [hunk 3]
```diff
@@ -2582,28 +2583,29 @@ void *syncWriteProgressive(threadInfo *pThreadInfo) {
...
             }
 
             pThreadInfo->totalAffectedRows += affectedRows;
 
             int currentPercent =
                 (int)(pThreadInfo->totalAffectedRows * 100 / totalRows);
             if (currentPercent > percentComplete) {
-                printf("[%d]:%d%%\n", pThreadInfo->threadID, currentPercent);
-                percentComplete = currentPercent;
+              printf("[%d]:%d%%\n", pThreadInfo->threadID, currentPercent);
```
Comment (sangshuduo) [inline src/demoInsert.c:2595]: indent

File: src/demoInsert.c [hunk 4]
```diff
@@ -2582,28 +2583,29 @@ void *syncWriteProgressive(threadInfo *pThreadInfo) {
             pThreadInfo->totalDelay += delay;
 
             if (affectedRows < 0) {
-                errorPrint("affected rows: %d\n", affectedRows);
-                goto free_of_progressive;
+              errorPrint("affected rows: %d\n", affectedRows);
```
Comment (sangshuduo) [inline src/demoInsert.c:2586]: indent

File: src/demoInsert.c [hunk 5]
```diff
@@ -2497,74 +2490,82 @@ void *syncWriteProgressive(threadInfo *pThreadInfo) {
...
-                } else {
-                    generated = generateProgressiveDataWithoutStb(
-                        tableName,
-                        /*  tableSeq, */
-                        pThreadInfo, pstr, insertRows, i, start_time,
-                        /* &(pThreadInfo->samplePos), */
-                        &remainderBufLen);
-                }
+              if (g_args.iface == STMT_IFACE) {
+                generated = prepareStmtWithoutStb(pThreadInfo, tableName, g_args.reqPerReq, insertRows, i, start_time);
```
Comment (sangshuduo) [inline src/demoInsert.c:2546]: indent

File: src/demoInsert.c [hunk 6]
```diff
@@ -2497,74 +2490,82 @@ void *syncWriteProgressive(threadInfo *pThreadInfo) {
...
+              } else {
+                generated = generateProgressiveDataWithoutStb(tableName,
```
Comment (sangshuduo) [inline src/demoInsert.c:2548]: indent

File: src/demoInsert.c [hunk 7]
```diff
@@ -2497,74 +2490,82 @@ void *syncWriteProgressive(threadInfo *pThreadInfo) {
     for (uint64_t tableSeq = pThreadInfo->start_table_from;
          tableSeq <= pThreadInfo->end_table_to; tableSeq++) {
         int64_t start_time = pThreadInfo->start_time;
-
+        char *  pstr = pThreadInfo->buffer;
+        uint64_t len = 0;
+        int32_t generated;
         for (uint64_t i = 0; i < insertRows;) {
+          if (g_args.pressure_mode) {
```
Comment (sangshuduo) [inline src/demoInsert.c:2497]: indent

**Review by sangshuduo — CHANGES_REQUESTED** [review]

File: inc/demo.h [hunk 8]
```diff
@@ -459,8 +461,8 @@ typedef struct SDataBase_S {
 } SDataBase;
 
 typedef struct SDbs_S {
-    char               cfgDir[MAX_FILE_NAME_LEN];
-    char               host[MAX_HOSTNAME_SIZE];
+    char                cfgDir[MAX_FILE_NAME_LEN];
```
Comment (sangshuduo) [inline inc/demo.h:464]: align

File: inc/demo.h [hunk 9]
```diff
@@ -523,17 +525,17 @@ typedef struct SuperQueryInfo_S {
 } SuperQueryInfo;
 
 typedef struct SQueryMetaInfo_S {
-    char               cfgDir[MAX_FILE_NAME_LEN];
-    char               host[MAX_HOSTNAME_SIZE];
-    uint16_t           port;
+    char                cfgDir[MAX_FILE_NAME_LEN];
```
Comment (sangshuduo) [inline inc/demo.h:528]: align

File: inc/demo.h [hunk 10]
```diff
@@ -523,17 +525,17 @@ typedef struct SuperQueryInfo_S {
...
+    uint16_t            port;
     struct sockaddr_in *serv_addr;
-    char               user[MAX_USERNAME_SIZE];
-    char               password[SHELL_MAX_PASSWORD_LEN];
-    char               dbName[TSDB_DB_NAME_LEN];
-    char               queryMode[SMALL_BUFF_LEN];  // taosc, rest
-    SpecifiedQueryInfo specifiedQueryInfo;
-    SuperQueryInfo     superQueryInfo;
-    uint64_t           totalQueried;
+    char                user[MAX_USERNAME_SIZE];
```
Comment (sangshuduo) [inline inc/demo.h:532]: align

**Review by sangshuduo — APPROVED** [review]
still few lines align mistake. not big deal but please correct them.

**Review by sangshuduo — APPROVED** [review]
my mistake. LGTM

**Pull request merged**